internal/ethapi/api.go

```diff
@@ -734,9 +727,9 @@ func (s *PublicBlockChainAPI) TraceCall(ctx context.Context, args CallArgs, bloc
 			ReturnValue: fmt.Sprintf("%x", ret),
 			StructLogs:  FormatLogs(vmenv.StructLogs()),
 		}, nil
+	} else {
+		return nil, err
 	}
```

Comment:
Similarly, there's no need for an else branch here, please drop and revert to the original code.